osh/word_parse.py

```diff
@@ -303,8 +305,8 @@ def _ReadVarOpArg(self, arg_lex_mode):
         return w
 
     def _ReadVarOpArg2(self, arg_lex_mode, eof_type, empty_ok):
-        # type: (lex_mode_t, Id_t, bool) -> CompoundWord
-        """Return a CompoundWord.
+        # type: (lex_mode_t, Id_t, bool) -> word_t
```

Comment:
OK I see why  many of these were changed to return word_t instead of CompoundWord

With some of them, a redirect is impossible, e.g. for `${x//</y}`, the lexer switches modes

```
(C
  (w <Lit_Chars echo>)
  (w
    (BracedVarSub
      left:<Left_DollarBrace "${">
      name_tok:<VSub_Name x>
      var_name:x
      suffix_op:(suffix_op.PatSub pat:(w <Lit_Other "<">) replace:(w <Lit_Chars y>) replace_mode:Lit_Slash slash_tok:<VOp2_Slash />)
      right:<Right_DollarBrace "}">
    )
  )
)

```

we probably need some asserts and then type narrowing, let me see